mooncaker/external_tools/mooncaker_bot.py

```diff
@@ -0,0 +1,149 @@
+        if update.effective_user.username in whitelist.split():
+            return dispatcher(update=update, context=context)
+        else:
+            update.message.reply_sticker(self.NAME_TO_STICKER[self.SAD_ZOE])
+            update.message.reply_text("Sorry, you cannot do that")
+
+    def start(self, update: Update, context: CallbackContext) -> None:
+        """Send a message when the command /start is issued."""
+        print(update.effective_chat.id)
+        user = update.effective_user
+        update.message.reply_sticker(self.NAME_TO_STICKER[self.LOLI])
+        update.message.reply_markdown_v2(
+            fr'Hi {user.mention_markdown_v2()} onii\-chan\!',
+            reply_markup=ForceReply(selective=True),
+        )
+        keyboard = [
+            [
+                InlineKeyboardButton("Command list", callback_data='1'),
+                InlineKeyboardButton("Cheer onii-chan", callback_data='2'),
+            ]
+        ]
+
+        reply_markup = InlineKeyboardMarkup(keyboard)
+
+        update.message.reply_text('How can I help you?', reply_markup=reply_markup)
+
+    def button(self, update: Update, context: CallbackContext) -> None:
+        """Parses the CallbackQuery and updates the message text."""
+        query = update.callback_query
+        query.answer()
+        if query.data == '1':
+            query.edit_message_text(
+                text=f"Available commands: \n /get_full_log \n /get_last_10_log \n /set_api_key \n /get_ReDiTi \n /get_count")
+        else:
+            query.edit_message_text(text=f"You are the best, onii-chan! I will always support you!")
+
+    def help_command(self, update: Update, context: CallbackContext) -> None:
+        """Send a message when the command /help is issued."""
+        update.message.reply_text('Help!')
+
+    def get_log(self, update: Update, context: CallbackContext, log_filename, lines=None) -> None:
+        with open(log_filename) as log:
+            if not lines is None:
+                with open("tmp.txt", "a") as less_log:
+                    for line in (log.readlines()[-lines:]):
+                        less_log.write(line + "\n")
+                with open("tmp.txt", "r") as less_log:
+                    context.bot.send_document(chat_id=update.effective_chat.id, document=less_log,
+                                              filename="last_10_lines_of_log.txt")
+                remove("tmp.txt")
+            else:
+                context.bot.send_document(chat_id=update.effective_chat.id, document=log,
+                                          filename="full_log.txt")
+
+    def set_api_key_req(self, update: Update, context: CallbackContext) -> int:
+        update.message.reply_text(
+            f"Please gimme all your new API key, I want it so bad..or use /cancel to make me stop waiting..")
+        return self.WAITING_API
+
+    def set_new_api(self, update: Update, context: CallbackContext, set_api):
```

Comment:
set_api is saved inside object, no need to pass it

Reply:
yup, forgot to use the object variable when converting the bot to a class 😢